com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Adjustment/HueNode.cs

```diff
@@ -0,0 +1,126 @@
+using System.Collections.Generic;
+using Usage = UnityEditor.ShaderGraph.GraphDelta.GraphType.Usage;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+    internal class HueNode : IStandardNode
+    {
+        static string Name = "Hue";
+        static int Version = 1;
+        public static NodeDescriptor NodeDescriptor => new(
+            Version,
+            Name,
+            new FunctionDescriptor[] {
+                new(
+                    1,
+                    "Degrees",
+@"
+    temp1.rg = In.bg;
+    temp1.ba = K.wz;
+    temp2.rg = In.gb;
+    temp2.ba = K.xy;
+    P = lerp(temp1, temp2, step(In.b, In.g));
+    temp1.xyz = P.xyw;
+    temp1.w = In.r;
+    temp2.r = In.r;
+    temp2.gba = P.yzx;
+    Q = lerp(temp1, temp2, step(P.x, In.r));
+    D = Q.x - min(Q.w, Q.y);
+    hsv.x = abs(Q.z + (Q.w - Q.y)/(6.0 * D + E));
+	hsv.y = D / (Q.x + E);
+	hsv.z = (D == 0) ? Q.x : (Q.x + E);
+    hue = hsv.x + Offset / 360;
+    hsv.x = (hue < 0)
+            ? hue + 1
+            : (hue > 1)
+                ? hue - 1
+                : hue;
+    // HSV to RGB
+    Out = hsv.z * lerp(K2.xxx, saturate((abs(frac(hsv.xxx + K2.xyz) * 6.0 - K2.www)) - K2.xxx), hsv.y);
+",
+                    new ParameterDescriptor("In", TYPE.Vec3, Usage.In),
+                    new ParameterDescriptor("Offset", TYPE.Float, Usage.In),
+                    new ParameterDescriptor("temp1", TYPE.Vec4, Usage.Local),
+                    new ParameterDescriptor("temp2", TYPE.Vec4, Usage.Local),
+                    new ParameterDescriptor("K", TYPE.Vec4, Usage.Local, new float[] { 0.0f, -0.333333f, 0.666667f, -1f }),
```

Comment:
Oh wow, I didn't know it was possible to use math as a static value default.  I converted them because I didn't know you could do (float)(- 1.0 / 3.0).  I'll change it to that!